uint/src/uint.rs

```diff
@@ -1462,6 +1462,23 @@ macro_rules! construct_uint {
 			}
 		}
 
+		// We implement `Eq` and `Hash` manually to workaround
+		// https://github.com/rust-lang/rust/issues/61415
+		impl $crate::core_::cmp::PartialEq for $name {
+			fn eq(&self, other: &$name) -> bool {
+				self.as_ref() == other.as_ref()
+			}
+		}
+
+		impl $crate::core_::cmp::Eq for $name {}
+
+		impl $crate::core_::hash::Hash for $name {
+			fn hash<H: $crate::core_::hash::Hasher>(&self, state: &mut H) {
+				// use the impl as slice &[u64]
+				<[u64] as $crate::core_::hash::Hash>::hash(&self.0[..], state);
```

Comment:
just `self.as_ref().hash(state);` ?

Reply:
Yeah, I think that would work

Reply:
yep, I wanted to be explicit here, but given the comment above it's ok to use `as_ref()` here, will fix this soon